src/ModelContextProtocol/Protocol/Types/CallToolRequestParams.cs

```diff
@@ -16,5 +18,5 @@ public class CallToolRequestParams : RequestParams
     /// Optional arguments to pass to the tool.
     /// </summary>
     [System.Text.Json.Serialization.JsonPropertyName("arguments")]
-    public Dictionary<string,object>? Arguments { get; init; }
+    public Dictionary<string, JsonElement>? Arguments { get; init; }
```

Comment:
The goal here is to force someone to do their own serialization?

Reply:
It is to enforce serialization of user-defined objects at the user level as opposed to the protocol level.